src/modules/gfx_maze.c

```diff
@@ -0,0 +1,234 @@
+// Simple maze animation.
+// Maze algorithm taken from: https://github.com/Figglewatts/mazegen
+// Thank you kind stranger!
+//
+// Copyright (c) 2019, Cyrill "xermic" Leutwiler <me@09f9.org>
+// 
+// Permission to use, copy, modify, and/or distribute this software for any
+// purpose with or without fee is hereby granted, provided that the above
+// copyright notice and this permission notice appear in all copies.
+// 
+// THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+// WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF
+// MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+// ANY SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+// WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN AN
+// ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+// OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+
+#include <types.h>
+#include <matrix.h>
+#include <timers.h>
+#include <stddef.h>
+
+#include <random.h>
+
+#include <stdlib.h>
+#include <stdbool.h>
+#include <limits.h>
+
+
+#define FPS 30
+#define FRAMETIME (T_SECOND / FPS)
+#define FRAMES (TIME_LONG * FPS)
+
+#define INVERSE_COLORS false
+
+#define NUM_DIRS 4
+typedef struct Stack Stack; // forward declaration
+extern int directions[];
+int directions[] = { 0, 1, 2, 3 };
+static int* maze;
+static Stack* x_stack;
+static Stack* y_stack;
+static int mx;
+static int my;
+
+static RGB wall = RGB(0, 0, 0);
+static int modno;
+static int frame;
+static ulong nexttick;
+
+typedef struct Stack {
+	int top;
+	unsigned capacity;
+	int* array;
+} Stack;
+
+Stack* stack_create(unsigned capacity) {
+	Stack* stack = (Stack*) malloc(sizeof(Stack));
+	stack->capacity = capacity;
+	stack->top = -1;
+	stack->array = (int*) malloc(stack->capacity * sizeof(int));
+	return stack;
+}
+
+bool stack_isfull(Stack* stack) {
```

Comment:
To help the compiler a bit, mark all these functions `static` and the small functions `static inline`.
This will omit the function call.

Reply:
ok